newIDE/app/src/ProjectManager/index.js

```diff
@@ -235,1135 +572,912 @@ const ProjectManager = React.memo(
+        editName(externalLayoutItemId);
       },
-      [copyExternalLayout, pasteExternalLayout]
+      [project, onProjectItemModified, editName, scrollToItem]
     );
 
-    const moveUpExternalLayout = React.useCallback(
-      (index: number) => {
-        if (index <= 0) return;
+    const onTreeModified = React.useCallback(
+      (shouldForceUpdateList: boolean) => {
+        if (unsavedChanges) unsavedChanges.triggerUnsavedChanges();
 
-        project.swapExternalLayouts(index, index - 1);
-        onProjectItemModified();
+        if (shouldForceUpdateList) forceUpdateList();
+        else forceUpdate();
       },
-      [project, onProjectItemModified]
+      [forceUpdate, forceUpdateList, unsavedChanges]
     );
 
-    const moveDownExternalLayout = React.useCallback(
-      (index: number) => {
-        if (index >= project.getExternalLayoutsCount() - 1) return;
-
-        project.swapExternalLayouts(index, index + 1);
-        onProjectItemModified();
-      },
-      [project, onProjectItemModified]
+    // Initialize keyboard shortcuts as empty.
+    // onDelete callback is set outside because it deletes the selected
+    // item (that is a props). As it is stored in a ref, the keyboard shortcut
+    // instance does not update with selectedItems changes.
+    const keyboardShortcutsRef = React.useRef<KeyboardShortcuts>(
+      new KeyboardShortcuts({
+        shortcutCallbacks: {},
+      })
     );
-
-    const copyEventsFunctionsExtension = React.useCallback(
-      (eventsFunctionsExtension: gdEventsFunctionsExtension) => {
-        Clipboard.set(EVENTS_FUNCTIONS_EXTENSION_CLIPBOARD_KIND, {
-          eventsFunctionsExtension: serializeToJSObject(
-            eventsFunctionsExtension
-          ),
-          name: eventsFunctionsExtension.getName(),
-        });
+    React.useEffect(
+      () => {
+        if (keyboardShortcutsRef.current) {
+          keyboardShortcutsRef.current.setShortcutCallback('onDelete', () => {
+            if (selectedItems.length > 0) {
+              deleteItem(selectedItems[0]);
+            }
+          });
+          keyboardShortcutsRef.current.setShortcutCallback('onRename', () => {
+            if (selectedItems.length > 0) {
+              editName(selectedItems[0].content.getId());
+            }
+          });
+          keyboardShortcutsRef.current.setShortcutCallback('onCopy', () => {
+            if (selectedItems.length > 0) {
+              selectedItems[0].content.copy();
+            }
+          });
+          keyboardShortcutsRef.current.setShortcutCallback('onPaste', () => {
+            if (selectedItems.length > 0) {
+              selectedItems[0].content.paste();
+            }
+          });
+          keyboardShortcutsRef.current.setShortcutCallback('onCut', () => {
+            if (selectedItems.length > 0) {
+              selectedItems[0].content.cut();
+            }
+          });
+        }
       },
-      []
+      [editName, selectedItems]
```

Comment:
This callback changes each time selected items change, so quite frequently. I don't know if there is a better way to do it regarding this optimization